File: daily-coding/coding-bat/python/pos_neg.py
# Given 2 int values, return True if one is negative and one is positive. Except if the parameter "negative" is True, then return True only if both are negative.

def pos_neg(a, b, negative):
  result = False
  if negative:
    if (a < 0) and (b < 0):
      result = True
  elif (a < 0 and b > -1):
    result = True
  elif (a > -1 and b < 0):
    result = True

  return result

File: daily-coding/coding-bat/java/posNeg.java
// Given 2 int values, return true if one is negative and one is positive. Except if the parameter "negative" is true, then return true only if both are negative.

public boolean posNeg(int a, int b, boolean negative) {
  boolean match = false;
  if (!negative) {
    if ((a < 0 && b > 0) || (a > 0 && b < 0)) {
      match = true;
    }
  } else if (a < 0 && b < 0) {
    match = true;
  }
  return match;
}
